presto-main/src/main/java/com/facebook/presto/execution/ParameterCollector.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.execution;
+
+import com.facebook.presto.sql.tree.DefaultTraversalVisitor;
+import com.facebook.presto.sql.tree.Parameter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ParameterCollector
```

Comment:
Since this class is never used in a place where a visitor is needed, I'd restructure it to work like `DependencyExtractor`. I.e., make it a utility class with a static method named `extractParameters` or similar and hide the visitor-ness inside an inner class.